chrome/content/zotero/xpcom/file.js

```diff
@@ -603,7 +603,33 @@ Zotero.File = new function(){
 			iterator.close();
 		}
 	}
-	
+
+	/**
+	 * Retrieve the size of the given file.
+	 *
+	 * Note: The returned size may be zero if the file does not exist or is a broken symlink
+	 *
+	 * @param path
+	 * @returns {Promise<integer>}
+	 */
+	this.getFileSize = async function (path) {
+		return OS.File.stat(path)
+			.then(
+				function (info) {
+					return info.size;
+				},
+				function (e) {
+					// Can happen if there's a symlink to a missing file
+					if (e instanceof OS.File.Error && e.becauseNoSuchFile) {
+						return 0;
```

Comment:
I was trying to abstract a bit more and save copying this try/catch everywhere we use this function. Since Zotero never knows whether a file exists on disk without checking, it made sense to me to abstract out that one error. It just seemed like a nicer interface to deal with, having the function handle that specific case. Otherwise, I envisioned writing that same try/catch every we use this function because you probably care about that exception, but none of the other (can't read disk, etc.) exceptions that can get handled a higher level since they are more serious.

But, that's an assumption which can always be wrong It's only one line without the exception handling so I'll just remove the whole function and each calling function can decide what it needs.